cmd/restic/cmd_stats.go

```diff
@@ -282,9 +308,15 @@ func verifyStatsInput(gopts GlobalOptions, args []string) error {
 // to collect information about it, as well as state needed
 // for a successful and efficient walk.
 type statsContainer struct {
-	TotalSize      uint64 `json:"total_size"`
-	TotalFileCount uint64 `json:"total_file_count"`
-	TotalBlobCount uint64 `json:"total_blob_count,omitempty"`
+	TotalSize                            uint64  `json:"total_size"`
+	TotalUncompressedSize                uint64  `json:"total_uncompressed_size,omitempty"`
+	TotalCompressedBlobsSize             uint64  `json:"total_compressed_blobs_size,omitempty"`
+	TotalCompressedBlobsUncompressedSize uint64  `json:"total_compressed_blobs_uncompressed_size,omitempty"`
+	CompressionRatio                     float64 `json:"compression_ratio,omitempty"`
+	CompressionProgress                  float64 `json:"compression_progress,omitempty"`
+	CompressionSpaceSaving               float64 `json:"compression_space_saving,omitempty"`
```

Comment:
`CompressionSpaceSaving` is somewhat redundant. As the JSON is intended for consumption by scripts, we can just let those calculate it themselves. My suggestion would be to remove the field from the struct and only calculate it on demand for the text output.

Reply:
I'm not sure about this, personally I'd expect the JSON output to include everything that's available in the text output so it feels more natural to have the compression space saving too.

I can certainly remove it if you feel strongly about it.

Reply:
Nah, I see the point. It's just that the PR started with adding 1 or 2 new fields, and now we have 4...